### PR TITLE
Support multiple tags in imagesteam

### PR DESCRIPTION
**Pull request by duyanyan**

### CODE REVIEW

File: ocp_resources/image_stream.py [hunk 1]
```diff
@@ -47,14 +47,18 @@ def to_dict(self):
...
+                        {
+                            "from": {
+                                "kind": "DockerImage",
+                                "name": self.image_repository,
+                            },
+                            "importPolicy": {"scheduled": self.scheduled},
+                            "name": self.tag,
+                            "referencePolicy": {"type": "Source"},
+                        }
+                    ],
```
Comment (myakove) [inline ocp_resources/image_stream.py:60]: please do 
```
if self.scheduled is not None:
            res["spec"]["tags"]["importPolicy"] = self.scheduled
```

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/image_stream.py (hunk 1 above)
Comment (duyanyan) [reply ocp_resources/image_stream.py:60]: I'm afraid we have to set it in default false, I missed ["scheduled"] which will cause schedule not working well:
KeyError: 'importPolicy'

Comment (myakove) [reply ocp_resources/image_stream.py:60]: Must we have `scheduled` under tags always?

Comment (duyanyan) [reply ocp_resources/image_stream.py:60]: scheduled is not a must value if we don't need it, but when we set it to false, it is same as without it like:
`      importPolicy: {} `

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/image_stream.py (hunk 1 above)
Comment (duyanyan) [reply ocp_resources/image_stream.py:60]: Fixed. Please review again.

Comment (rnetser) [reply ocp_resources/image_stream.py:60]:
```
Also tags may have multiple tags but we limit the user to set only one?
```
?

Comment (duyanyan) [reply ocp_resources/image_stream.py]: Updated the code, please check again @rnetser @myakove

File: ocp_resources/image_stream.py [hunk 2]
```diff
@@ -43,18 +43,21 @@ def to_dict(self):
...
-                    "tags": {
-                        "from": {"kind": "DockerImage", "name": self.image_repository},
-                        "name": self.tag,
-                        "referencePolicy": {"type": "Source"},
-                    },
-                }
+        tags = []
+        tag_name = self.tag
+        repository_name = self.image_repository
+        for x, y in zip(tag_name, repository_name):
```
Comment (myakove) [inline ocp_resources/image_stream.py:49]: I think the user should pass the tags dict.

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/image_stream.py (hunk 2 above)
Comment (duyanyan) [reply ocp_resources/image_stream.py]: Fixed.

File: ocp_resources/image_stream.py [hunk 3]
```diff
@@ -14,8 +14,6 @@ def __init__(
         name=None,
         namespace=None,
         client=None,
-        image_repository=None,
-        scheduled=None,
         lookup_policy=False,
         tag=None,
```
Comment (rnetser) [inline ocp_resources/image_stream.py:18]:
```suggestion
        tags=None,
```

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/image_stream.py (hunk 3 above)
Comment (duyanyan) [reply ocp_resources/image_stream.py]: Fixed

**Review by rnetser — APPROVED** [review]

**Comment by duyanyan** [comment]
!verified

File: ocp_resources/image_stream.py [hunk 4]
```diff
@@ -34,27 +32,14 @@ def __init__(
...
-                "spec": {
-                    "lookupPolicy": {"local": self.lookup_policy},
-                    "tags": {
-                        "from": {"kind": "DockerImage", "name": self.image_repository},
-                        "name": self.tag,
-                        "referencePolicy": {"type": "Source"},
-                    },
-                }
-            }
+            {"spec": {"lookupPolicy": {"local": self.lookup_policy}, "tags": self.tags}}
```
Comment (myakove) [inline ocp_resources/image_stream.py:43]: ```"tags": self.tags}```
Is this OK to pass tags: None? Or do we need to do:

```
if tags:
```

Comment (duyanyan) [reply ocp_resources/image_stream.py:43]: no, tag should be defined, or else the image stream doesn't work well

**Review by myakove — APPROVED** [review]

**Comment by myakove** [comment]
@duyanyan Please rebase

**Comment by duyanyan** [comment]
> @duyanyan Please rebase

Done

**Comment by myakove** [comment]
!verified

**Comment by myakove** [comment]
!verified

**Pull request merged**